perfzero/lib/cloud_manager.py

```diff
@@ -133,9 +134,37 @@ def create(username, project, zone, machine_type, accelerator_count,
   logging.info('Successfully created gcloud computing instance %s with %s accelerator.\n',  # pylint: disable=line-too-long
                instance_name, accelerator_count)
 
+  # Wait until we can ssh to the newly created computing instance
+  cmd = 'gcloud compute ssh {} --project={} --zone={} --strict-host-key-checking=no --command="ls"'.format(instance_name, project, zone)  # pylint: disable=line-too-long
```

Comment:
Can we just run the `git clone` and `usermod -aG docker username` commands here, instead of a dummy `ls` command?

Reply:
One benefit with the current implementation is that, we are pretty sure that `ls` will not fail. Thus the command will only fail if `ssh` failed. On the other hand, if we do `git clone` and `usermod ...` and the command failed, it will not be clear whether failure is due to ssh connection. Note that we only want to retry if it is due to ssh failure.

Another benefit with the current implementation is that it is more readable. The code is making it clear that we expect ssh connection to possibility fail, and if it fails, we are willing to retry for a given number of time. And after it succeeds, we will run to command with no tolerance for failure. The alternative approach makes it a bit hard to express this logic.